compiler/rustc_middle/src/ty/util.rs

```diff
@@ -1224,6 +1224,13 @@ impl<'tcx> Ty<'tcx> {
         return false;
     }
 
+    pub fn is_thread_local(self, tcx: TyCtxt<'tcx>) -> bool {
+        if let ty::Adt(adt_def, ..) = self.kind() {
+            return tcx.get_diagnostic_item(sym::LocalKey).map_or(false, |t| adt_def.did() == t);
+        }
+        return false;
```

Comment:
I think you'll get a clippy warning telling you "just use `false` here? If so, please force push a fix.

Reply:
Ah yeah. I don't think is setup to warn for this kind of thing on rustc, but I'll force push a fix anyway. For some reason just returning `true/false` without a `return` keyword and semicolon looks wrong to me, but it's not a hill I'll die on!